src/nplinker/genomics/antismash/downloader.py

```diff
@@ -0,0 +1,85 @@
+import os
+import shutil
+from nplinker.logconfig import LogConfig
+from nplinker.utils import download_and_extract_archive
+from nplinker.utils import list_dirs
+from nplinker.utils import list_files
+
+
+logger = LogConfig.getLogger(__name__)
+
+# urls to be given to download antismash data
+ANTISMASH_DB_PAGE_URL = 'https://antismash-db.secondarymetabolites.org/output/{}/'
+ANTISMASH_DB_DOWNLOAD_URL = 'https://antismash-db.secondarymetabolites.org/output/{}/{}'
+
+# The antiSMASH DBV2 is for the availability of the old version, better to keep it.
+ANTISMASH_DBV2_PAGE_URL = 'https://antismash-dbv2.secondarymetabolites.org/output/{}/'
+ANTISMASH_DBV2_DOWNLOAD_URL = 'https://antismash-dbv2.secondarymetabolites.org/output/{}/{}'
+
+
+def _check_roots(download_root: str, extract_root: str):
+    if download_root == extract_root:
+        raise ValueError(
+            "Identical path of download directory and extract directory")
+
+
+def _check_extract_path(extract_path: str):
+    if os.path.exists(extract_path):
+        # check if extract_path is empty
+        files = list(os.listdir(extract_path))
+        if len(files) != 0:
+            raise ValueError(f'Nonempty directory: "{extract_path}"')
+    else:
+        os.makedirs(extract_path, exist_ok=True)
+
+
+def download_and_extract_antismash_metadata(refseq_assembly_id: str,
+                                            download_root: str,
```

Comment:
For public methods,  Helge and me agreed before that path should support a type  `str | Pathlike` (`os.Pathlike`).  The path should be converted to `Pathlib.Path` object and the returned path should be a `str`.

Reply:
What do you mean here by the "returned path"? I added support for PathLike type in the private functions as well, otherwise their type hint would be incorrect now

Reply:
it means the `path_out` should be a `str`
```
def foo(path_in: str |  Pathlike) -> str:
    ...
    return path_out
```